### ISSUE TITLE
Block documentation does not list allowed values for variant prop or explain aria-label use.

### ISSUE THREAD

**Issue by danielck** [vrk-kpa/suomifi-ui-components#727]
## The Problem

The Block component has a `variant` prop which only supports specific values ('default','div','span','section','header','nav','main','footer'). This is not communicated clearly in the prop documentation. 

Also, e.g. `<section>` requires an aria-label to be interpreted as a landmark by many assistive technology, and `nav` elements should have aria-labels when there are multiple navigation blocks of differing type on the same page.

## Proposed Solution

- Rewrite the prop description of `variant` to explain that possible values are 'default','div','span','section','header','nav','main' and 'footer'.
- Make it explicit that attributes like aria-label, aria-labelledby etc can be used.
